pkg/storage/stores/tsdb/head_manager.go

```diff
@@ -151,8 +151,8 @@ func (m *HeadManager) loop() {
 			return nil
 		}
 
-		if err := m.buildTSDBFromWAL(m.prev.initialized); err != nil {
-			return errors.Wrap(err, "building tsdb head")
+		if err := m.buildTSDBFromHead(m.prevHeads); err != nil {
+			return err
```

Comment:
Let us retain the error wrapping since it helps add more context and debug the issue easily.
```suggestion
			return errors.Wrap(err, "building tsdb from head")
```